test/unit/contracts-component.spec.ts

```diff
@@ -0,0 +1,491 @@
+import { ICacheStorageComponent } from '@dcl/core-commons'
+import { ChainId, Network, NFTCategory } from '@dcl/schemas'
+import { createContractsComponent } from '../../src/ports/contracts/component'
+import { DBCollection, IContractsComponent } from '../../src/ports/contracts/types'
+import { IPgComponent } from '../../src/ports/db/types'
+import { SquidNetwork } from '../../src/types'
+import { createTestPgComponent } from '../components'
+import { createCacheMockedComponent } from '../mocks/cache-mock'
+
+let inMemoryCache: ICacheStorageComponent
+let mockGet: jest.MockedFn<ICacheStorageComponent['get']>
+let mockSet: jest.MockedFn<ICacheStorageComponent['set']>
+
+jest.mock('../../src/logic/chainIds', () => ({
+  getEthereumChainId: () => ChainId.ETHEREUM_SEPOLIA,
+  getPolygonChainId: () => ChainId.MATIC_AMOY,
+  getNetwork: (network: string) => (network === SquidNetwork.POLYGON ? Network.MATIC : Network.ETHEREUM),
+  getNetworkChainId: (network: string) => (network === SquidNetwork.POLYGON ? ChainId.MATIC_AMOY : ChainId.ETHEREUM_SEPOLIA)
+}))
+
+describe('when getting marketplace contracts', () => {
+  let contractsComponent: IContractsComponent
+  let pgComponent: IPgComponent
+
+  beforeEach(() => {
+    mockGet = jest.fn()
+    mockSet = jest.fn()
+
+    inMemoryCache = createCacheMockedComponent({
+      get: mockGet as any,
+      set: mockSet
+    })
+
+    pgComponent = createTestPgComponent({ query: jest.fn() })
+    contractsComponent = createContractsComponent({ dappsDatabase: pgComponent, inMemoryCache })
+  })
+
+  it('should include LAND, Estates, and Names contracts', () => {
+    const contracts = contractsComponent.getMarketplaceContracts()
+
+    expect(contracts.some(c => c.name === 'LAND')).toBe(true)
+    expect(contracts.some(c => c.name === 'Estates')).toBe(true)
+    expect(contracts.some(c => c.name === 'Names')).toBe(true)
+  })
+})
+
+describe('when getting collection contracts', () => {
+  let contractsComponent: IContractsComponent
+  let pgComponent: IPgComponent
+  let pgQueryMock: jest.Mock
+
+  beforeEach(() => {
+    mockGet = jest.fn()
+    mockSet = jest.fn()
+
+    inMemoryCache = createCacheMockedComponent({
+      get: mockGet as any,
+      set: mockSet
+    })
+  })
+
+  describe('and there are no collections', () => {
+    beforeEach(() => {
+      pgQueryMock = jest.fn().mockResolvedValue({ rows: [] })
+      pgComponent = createTestPgComponent({ query: pgQueryMock })
+      contractsComponent = createContractsComponent({ dappsDatabase: pgComponent, inMemoryCache })
```

Comment:
You can avoid the redefinition of the `contractsComponent` the `pgComponent` and even the `inMemoryCacfhe` components in each `beforeEach` by creating a root beforeEach that instantiates the components.
You can then use the `pgQeuryMock`, the `mockGet` or `mockSet` to set the different values in each `beforeEach`